### PR TITLE
Fix error code reported for context errors

### PR DESCRIPTION
**Pull request by bduffany**
---

**Version bump**: Patch 

**Related issues**: https://github.com/buildbuddy-io/buildbuddy-internal/issues/1321

### CODE REVIEW

**Comment by siggisim** [comment]
I'm a little nervous that we're currently retrying a lot of these errors (because they return to bazel as Unavailable) but now we won't be retrying them (because we return DeadlineExceeded).

File: enterprise/server/remote_execution/executor/executor.go [hunk 1]
```diff
@@ -349,11 +349,26 @@ func (s *Executor) ExecuteTaskAndStreamResults(ctx context.Context, task *repb.E
...
 	if err == nil {
 		alert.UnexpectedEvent("incomplete_command_with_nil_error")
 		return status.UnknownErrorf("Command did not complete, for unknown reasons (internal status %d)", exitCode)
 	}
+	// If the context timed out or was cancelled, ignore any error returned from
+	// the command since it's most likely caused by the context error, and we
+	// don't want to require container implementations to have to handle these
+	// cases explicitly.
+	if ctxErr := ctx.Err(); ctxErr != nil {
+		log.Infof("Ignoring command error likely caused by %s: %s", ctxErr, err)
```
Comment (tylerwilliams) [inline enterprise/server/remote_execution/executor/executor.go:362]: can you make sure this is a sufficient log level for debugging this in prod?

**Review by tylerwilliams — APPROVED** [review]
This LGTM -- make sure @siggisim is cool with it before submitting.

We can babysit it carefully / yank if necessary.

**Review by vadimberezniker — APPROVED** [review]

**Review by siggisim — APPROVED** [review]
Thanks!!

**Pull request merged**